src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/DocumentPresentation/TextDocumentTextPresentationEndpointTests.cs

```diff
@@ -36,7 +36,8 @@ public async Task Handle_Html_MakesRequest()
         var clientConnection = new Mock<IClientConnection>(MockBehavior.Strict);
         clientConnection
             .Setup(l => l.SendRequestAsync<IRazorPresentationParams, WorkspaceEdit?>(CustomMessageNames.RazorTextPresentationEndpoint, It.IsAny<IRazorPresentationParams>(), It.IsAny<CancellationToken>()))
-            .ReturnsAsync(response);
+            .ReturnsAsync(response)
+            .Verifiable();
```

Comment:
Fun fact, without this change, this test wasn't actually asserting anything. More fun fact, I didn't even need to modify the test below, it passed regardless.

Mocks are fun!

Reply:
To make sure I understand - this makes it so that ```.verify``` will actually verify all calls that were setup were made? Or that any calls that weren't setup weren't made? Or a combination of both? :) 

And the test below technically does the same thing, but and since we didn't setup any calls, it simply verifies no calls were made?

I would expect the test you modified below would throw something like "MethodNotFound" (or is it "MethodNotImplemented") exception if it's actually called before we even got to verify(), so verify is basically a no-op()?

Reply:
OK, read through https://codingcanvas.com/moq-setup-verify-verifiable-callback-explained/, I think I understand what's going on (assuming that article is right :) ). 

But basically in the second test ```Handle_CSharp_DoesNotMakeRequest```, ```verify``` is a no-op since nothing was setup from what I understand. You can just delete it and the test will work as it does today - it will throw ```methodnotfound``` exception if the code is changed to actually call C#. It will pass if call to C# isn't mae.

Reply:
also the other way is to fix the first test could have been calling VerifyAll at the end, which should verify all setups even if not explicitly marked as verifiable (if my understanding is correct)

Reply:
You read an article, so you know more than me!

But yes, you are correct. My expectations with this PR is that if we start calling Roslyn again, the below test will fail with a "Setup missing for method" or whatever Moq says. `VerifyAll` is a good tip though, seems like I could have done that here instead.